gomock/matchers.go

```diff
@@ -403,15 +410,20 @@ func Not(x any) Matcher {
 	return notMatcher{Eq(x)}
 }
 
-// Regex checks whether a string parameter matches the associated regex.
+// Regex checks whether parameter matches the associated regex.
 //
 // Example usage:
 //
 // Regex("[0-9]{2}:[0-9]{2}").Matches("23:02") // returns true
+// Regex("[0-9]{2}:[0-9]{2}").Matches([]byte{50,51,58,48,50}) // returns true
 // Regex("[0-9]{2}:[0-9]{2}").Matches("hello world") // returns false
-// Regex("[0-9]{2}").Matches(21) // returns false as it's not a string
+// Regex("[0-9]{2}").Matches(21) // returns false as it's not a valid type
 func Regex(regexStr string) Matcher {
-	return regexMatcher{regex: regexStr}
+	compiledRegexp, err := regexp.Compile(regexStr)
+	if err == nil {
+		return regexMatcher{regex: compiledRegexp}
+	}
+	return regexMatcher{regex: nil, compileErr: err}
```

Comment:
```suggestion
	return regexMatcher{regex: regexp.MustCompile(regexStr)}
```
_(This is together with the previous suggested change.)_